exchange/exchangetest/firstpartydata-multibidder-config-valid.json

```diff
@@ -0,0 +1,227 @@
+{
+  "requestType": "openrtb2-web",
+  "incomingRequest": {
+    "ortbRequest": {
+      "id": "some-request-id",
+      "site": {
+        "page": "test.somepage.com"
+      },
+      "imp": [
+        {
+          "id": "some-imp-id",
+          "banner": {
+            "format": [
+              {
+                "w": 600,
+                "h": 500
+              },
+              {
+                "w": 300,
+                "h": 600
+              }
+            ]
+          },
+          "ext": {
+            "appnexus": {
+              "placementId": 1
+            },
+            "rubicon": {
+              "accountId": 1,
+              "siteId": 2,
+              "zoneId": 3
+            }
+          }
+        }
+      ],
+      "ext": {
+        "prebid": {
+          "data": {
+            "bidders": [
+              "appnexus",
+              "rubicon"
+            ]
+          },
+          "bidderconfig": [
+            {
+              "bidders": [
+                "appnexus"
+              ],
+              "config": {
+                "ortb2": {
+                  "site": {
+                    "domain": "fpd_appnexus_site_domain",
+                    "page": "fpd_appnexus_site_page"
+                  }
+                }
+              }
+            },
+            {
+              "bidders": [
+                "rubicon"
+              ],
+              "config": {
+                "ortb2": {
+                  "site": {
+                    "domain": "fpd_rubicon_site_domain",
+                    "page": "fpd_rubicon_site_page"
+                  }
+                }
+              }
+            }
+          ]
+        }
+      }
+    }
+  },
+  "outgoingRequests": {
+    "appnexus": {
+      "expectRequest": {
+        "ortbRequest": {
+          "id": "some-request-id",
+          "imp": [
+            {
+              "id": "some-imp-id",
+              "banner": {
+                "format": [
+                  {
+                    "w": 600,
+                    "h": 500
+                  },
+                  {
+                    "w": 300,
+                    "h": 600
+                  }
+                ]
+              },
+              "ext": {
+                "bidder": {
+                  "placementId": 1
+                }
+              }
+            }
+          ],
+          "site": {
+            "domain": "fpd_appnexus_site_domain",
+            "page": "fpd_appnexus_site_page"
+          },
+          "ext": {
+            "prebid": {
+              "data": {
+                "eidpermissions": null,
+                "bidders": [
+                  "appnexus",
+                  "rubicon"
+                ]
+              },
+              "bidderconfig": [
```

Comment:
This data isn't typically sent in the request right? Is it in the bid requests here because of how something is mocked in this test?

Reply:
Good catch! During the implementation I decided this is the set up of unit test. But it's not. Because of change from BidRequest to RequestWrapper any ext changes has to be updated in both RequestWrapper and previously extracted `requestExt`. I added related code in exchange.go. Optimization: RequestWrapper and `requestExt` runs only in case if any fpd was extracted.